### PR TITLE
Add MSRV

### PR DESCRIPTION
**Pull request by rusty-snake**
Relates #135

- [x] add ci check
- [x] fix current code for 1.46
- [x] Add `rust-version` to `Cargo.toml`.
- [x] Add badged to README.md
- [x] Document MSRV bump policy
- [x] CHANGELOG.md

### CODE REVIEW

**Comment by rusty-snake** [comment]
```
error[E0658]: the `unsafe_op_in_unsafe_fn` lint is unstable
  --> libseccomp/src/lib.rs:56:1
   |
56 | #![deny(unsafe_op_in_unsafe_fn)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #71668 <https://github.com/rust-lang/rust/issues/71668> for more information
```

AFAIK this does not effect users of our crate (will test later) and could be disable via `cfg_attr` for the msrv run (AFAIK).

```
error[E0277]: `[(&str, action::ScmpAction); 9]` is not an iterator
   --> libseccomp/src/action.rs:117:21
    |
117 |         for data in test_data {
    |                     ^^^^^^^^^ borrow the array with `&` or call `.iter()` on it to iterate over it
    |
    = help: the trait `std::iter::Iterator` is not implemented for `[(&str, action::ScmpAction); 9]`
    = note: arrays are not iterators, but slices like the following are: `&[1, 2, 3]`
    = note: required by `std::iter::IntoIterator::into_iter`
```

Can we make `cargo check` ignore test code? Anyway we can just make this work for older rust versions w/o any real downside. This does not effect users too AFAIK.

File: .github/workflows/build-test.yaml [hunk 1]
```diff
@@ -108,6 +108,21 @@ jobs:
       - name: Build crate
         run: cargo build --target ${{ matrix.target }} --all-features
 
+  msrv:
+    name: MSRV
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Rust 1.46.0
```
Comment (ManaSugi) [inline .github/workflows/build-test.yaml:116]: Could you use a global env variable of msrv?
You can add `env` workflow between `on` and `jobs` workflow:
Ref: https://docs.github.com/en/actions/learn-github-actions/environment-variables#about-environment-variables

```yml
env:
  msrv: 1.46.0
```

Then, you can reference the version by `${{ env.msrv }}`.

File: libseccomp/build.rs [hunk 2]
```diff
@@ -3,13 +3,42 @@
...
+    }
+    if rustc_version[1] < 52 {
+        // unsafe_block_in_unsafe_fn
+        println!("cargo:rustc-cfg=msrv_compat_1_52");
+    }
+    if rustc_version[1] < 53 {
+        // array_into_iter_impl
+        println!("cargo:rustc-cfg=msrv_compat_1_53");
+    }
+
```
[the remark this answers is not in the document]
Comment (rusty-snake) [reply libseccomp/build.rs:41]: Do you want it as an new job? With which checks?

**Comment by rusty-snake** [comment]
```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `"Error { kind: Errno(EACCES), source: None, message: \"The library doesn\\\'t permit the particular operation\" }"`,
 right: `"Error { kind: Errno(EACCES), source: None, message: \"The library doesn\'t permit the particular operation\" }"`', libseccomp/src/error.rs:387:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: test failed, to rerun pass '-p libseccomp --lib'
```

File: libseccomp/build.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ManaSugi) [reply libseccomp/build.rs]: Yes, a new job is better I think.

File: .github/workflows/build-test.yaml [hunk 3]
```diff
@@ -108,6 +108,21 @@ jobs:
       - name: Build crate
         run: cargo build --target ${{ matrix.target }} --all-features
 
+  msrv:
+    name: MSRV
```
[the remark this answers is not in the document]
Comment (ManaSugi) [reply .github/workflows/build-test.yaml:116]: @rusty-snake Could you add this to be easy to understand the msrv version on PR screen?

[the remark this answers is not in the document]
Comment (ManaSugi) [reply .github/workflows/build-test.yaml:116]: Ah, sorry, you're right. we cannot do this

**Comment by rusty-snake** [comment]
There are some rust internal changes in 1.53.0 (nothing in rust/RELEASES.md) that cause `error.rs - test_debug` to fail on rust 1.52 or lower. For no the test is disabled on rust <1.53.

**Pull request merged**